### ISSUE TITLE
Question about bidirectional distillation

### ISSUE THREAD

**Issue by jsg921019** [tianweiy/CausVid#44]
What is the difference between configs/wan_bidirectional_dmd_from_scratch.yaml and configs/wan_bidirectional_dmd.yaml?
Which one should i use if i want to train it myself?

**Comment by tianweiy** [comment]
the from_scratch version has backward_simulation=False meaning that it is using the videos in the training dataset.

the other version uses backward simulation meaning that it takes a prompt from the training dataset and simulate the input video for training (instead of using a gt video). 

The second version probably performs better but it is slightly slower. Moreover, if you choose to use the second version, it might be helpful to use better prompts (e.g. from https://github.com/guandeh17/Self-Forcing/tree/main?tab=readme-ov-file#download-text-prompts-and-ode-initialized-checkpoint)

**Comment by jsg921019** [comment]
Thanks for fast and accurate reply. I am following your code, and got another question about this code.

https://github.com/tianweiy/CausVid/blob/ca2115039172dfa3a02deee1534408a09680f29f/causvid/dmd.py#L259-L265

I think the loss should be just grad**2? What is the underlying meaning of original_latent and (original_latent - grad)?

**Comment by jsg921019** [comment]
related: https://github.com/tianweiy/DMD2/issues/68#issue-2926677180

**Comment by thbupt** [comment]
@jsg921019 ‌According to the formula in the paper, the derivative of loss with respect to original_latent does ‌not‌ need to be computed via backpropagation through the real or fake networks(I think this is different from the gan loss). In fact, d_Loss/d_ original_latent ‌exactly corresponds to‌ grad in the code.

**Comment by jsg921019** [comment]
@thbupt thanks for the reply.
So this code is just a nice trick to attach gradient to the original_latent.

**Comment by aihao2000** [comment]
> the from_scratch version has backward_simulation=False meaning that it is using the videos in the training dataset.
> 
> the other version uses backward simulation meaning that it takes a prompt from the training dataset and simulate the input video for training (instead of using a gt video).
> 
> The second version probably performs better but it is slightly slower. Moreover, if you choose to use the second version, it might be helpful to use better prompts (e.g. from https://github.com/guandeh17/Self-Forcing/tree/main?tab=readme-ov-file#download-text-prompts-and-ode-initialized-checkpoint)

Hi, I'd like to ask why you think backward_simulation might be more effective. It seems to me that simulated samples using only four steps of backward_simulation are very different from real samples, exhibiting blurring and distortion. Do I need at least a GAN, like DMD2, to maintain image quality? In DMD2 ablation experiments, I noticed a significant drop in metrics without the GAN. For my task, real data is scarce, and using only backward simulation hasn't yielded satisfactory results. I'm looking into this issue and look forward to your response.
